core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/YoutubePlayerViewDefaultWebUITest.kt

```diff
@@ -0,0 +1,170 @@
+import com.pierfrancescosoffritti.androidyoutubeplayer.test.R
+import org.hamcrest.CoreMatchers.*
+import org.hamcrest.Matcher
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * This class tests if the player's default ui is the web-based ui.
+ * Additionally, it tests all 4 manual initialization methods to check
+ * if they create their expected UI elements.
+ */
+@LargeTest
+class YoutubePlayerViewDefaultWebUITest {
+
+    private lateinit var testActivity: Activity
+    private lateinit var youTubePlayerViewUsingInitializer1: YouTubePlayerView
+    private lateinit var youTubePlayerViewUsingInitializer2: YouTubePlayerView
+    private lateinit var youTubePlayerViewUsingInitializer3: YouTubePlayerView
+    private lateinit var youTubePlayerViewUsingInitializeWithNativeUi: YouTubePlayerView
+
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(TestActivity::class.java)
+
+    @Before
+    fun setup() {
+        testActivity = activityRule.activity
+        youTubePlayerViewUsingInitializer1 = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_1)
+        youTubePlayerViewUsingInitializer2 = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_2)
+        youTubePlayerViewUsingInitializer3 = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_3)
+        youTubePlayerViewUsingInitializeWithNativeUi = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_native)
+    }
+
+
+    /**
+     * Returns the FrameLayout of the Native UI for the given [youtubePlayerId] of a player.
+     * @param youtubePlayerId Resource id of a YouTubePlayerView
+     * @return ViewMatcher of Native UI FrameLayout
+     */
+    private fun getNativeUiViewGroup(youtubePlayerId: Int): Matcher<View> {
+        return allOf(
+                isDescendantOfA(
+                        withId(youtubePlayerId)
+                ),
+                withClassName(
+                        containsString(
+                                FrameLayout::class.java.canonicalName)
+                ),
+                withParent(
+                        withClassName(
+                                containsString(
+                                        LegacyYouTubePlayerView::class.java.canonicalName)
+                        )
+                )
+        )
+    }
+
+
+    @Test
+    fun testIfDefaultIsWebUi() {
+        // Prepare
+
+        // Act
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view)
+        ).check(matches(hasChildCount(0)))
+    }
+
+    @Test
+    fun testIfNativeViewIsSetByXMLAttribute() {
+        // Prepare
+
+        // Act
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view_native)
+        ).check(
+                matches(
+                        not(hasChildCount(0))
```

Comment:
Is there a more explicit way to check this? For example: we could give a name/id to the ViewGroup that contains the controls and check that it's not there, by looking for it's name/id.

Reply:
done

Reply:
Feel free to mark these as resolved when you do them.

Reply:
ok sure